DatadogInternal/Sources/NetworkInstrumentation/NetworkInstrumentationFeature.swift

```diff
@@ -92,7 +92,7 @@ internal final class NetworkInstrumentationFeature: DatadogFeature {
             interceptDidFinishCollecting: { [weak self] session, task, metrics in
                 self?.task(task, didFinishCollecting: metrics)
 
-                if #available(iOS 15, tvOS 15, *) {
+                if #available(iOS 15, tvOS 15, *), !task.dd.hasCompletion {
                     // iOS 15 and above, didCompleteWithError is not called hence we use task state to detect task completion
                     // while prior to iOS 15, task state doesn't change to completed hence we use didCompleteWithError to detect task completion
                     self?.task(task, didCompleteWithError: task.error)
```

Comment:
suggestion/ Don't we leak the abstraction here? It seems that `swizzler` implementation could perform this whole check. This way `NetworkInstrumentationFeature` won't be infected with OS-specific checks as it will rely on the `swizzler` to call `interceptDidCompleteWithError` uniformly, no matter of `task` state or OS specifics.

This complexity is much easier to unit test in downstream components. A broken test will be much shorter to troubleshoot on this or that OS version and testing different edge cases will be more isolated.

Reply:
it is matter of where to handle this kind of complexity. We have simpler swizzler, easy to unit test but complex usage which is covered with test cases.

I like this because complexity stays at one place rather distributing to all over.

If you feel strong about it with some clear benefit, we have to do a refactor which I think must not be part of this PR.

Reply:
No strong opinion 👍. If tests coverage is there, at this point it would only imply refactoring with no immediate gain.